src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1963,7 +1963,46 @@ public TryWriteInterpolatedStringHandler(int literalLength, int formattedCount,
             /// <summary>Writes the specified string to the handler.</summary>
             /// <param name="value">The string to write.</param>
             /// <returns>true if the value could be formatted to the span; otherwise, false.</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public bool AppendLiteral(string value)
+            {
+                // See comment on inlining and special-casing in DefaultInterpolatedStringHandler.AppendLiteral.
+
+                if (value.Length == 1)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)pos < (uint)destination.Length)
+                    {
+                        destination[pos] = value[0];
+                        _pos = pos + 1;
+                        return true;
+                    }
+
+                    return Fail();
+                }
+
+                if (value.Length == 2)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)(pos + 1) < (uint)destination.Length)
```

Comment:
Because this is introducing unsafe code we'd better make 100% sure _pos_ can never be negative. I don't see how this could ever happen in practice given that this is a ref struct and not subject to tearing or multithreading nonsense, but it'd be good to double-check.

Reply:
This is a holdover from when I was trying to get the JIT to eliminate bounds checks in the guarded block.  Now that we're using unsafe code to avoid them, I can change the condition to be:
```C#
if ((uint)pos < destination.Length - 1)
```
and that should eliminate any such concern, yes?